File: DataProcessing/queries.py
"""
A collection of functions for the key figures, uses the Json called from the DB.
(here a mock of how one looks like based off the xlxs_reader)

package installed, pymongo
"""
import xlxs_reader as xlxsr
import pymongo


def equity_ratio(request):
    """
    :param data:
    :return:
    """
    '''
    myclient = pymongo.MongoClient("mongodb://localhost:27017/")
    mydb = myclient["mydatabase"]
    mycol = mydb["entries"]

    myquery = {"company_id": request['company_id']}
    data = mycol.find(myquery)
    '''

    result = {
        "company_id": request['company_id'],
        "equity_ratios": []
    }
    data = xlxsr.write_json()

    for data_set in data:
        if data_set:
            equity = data_set["balance_sheet"]["passives"]["equity"]["total"]
            current_assets_total = data_set["balance_sheet"]["actives"]["current_assets"]["total"]
            fixed_assets_total = data_set["balance_sheet"]["actives"]["fixed_assets"]["total"]

            key_figure = equity / (current_assets_total + fixed_assets_total)

            result["equity_ratios"].append({
                "period" : data_set["period"],
                "key_figure" : key_figure
            })

    #return mock data results
    return result

#Fremdfinanzierungsgrad
#Selbstfinanzierungsgrad
#Intensität des Umflaufvermögen
#Intensität des Anlagevermögen
#Liquiditätsgrad 1 (cash Ratio)
#Liquiditätsgrad 2 (Quick Ratio)
#Liquiditätgrad 3 (Current ratio)
#Anlagedeckungsgrad 1
#Anlagedeckungsgrad 2
#Eigenkapitalrendite
#Gesamtkapitalrendite
#Umsatzrendite

def main():
    """
    main
    :return:
    """
    print( equity_ratio({"company_id":"example"}) )

if __name__ == "__main__":
    main()

File: DataProcessing/queries.js
const {MongoClient} = require('mongodb');


async function EquityRatio(request) {
    try {
        const uri = "mongodb://localhost:27017/";
        const client = new MongoClient(uri);
        await client.connect();

        const database = client.db("kennzahlen")
        const query = {"company_id": request.company_id};
        let data = database.collection('report').find(query)

        const result = {
            "company_id": request.company_id,
            "equity_ratios": []
        };

        await data.forEach((dataset) => {
            if (dataset){
                const equity = dataset.balance_sheet.passives.equity.total;
                const currentAssetsTotal = dataset.balance_sheet.actives.current_assets.total;
                const fixedAssetsTotal = dataset.balance_sheet.actives.fixed_assets.total;

                const keyFigure = equity / (currentAssetsTotal + fixedAssetsTotal);

                result.equity_ratios.push({
                    "period": dataset.period,
                    "key_figure": keyFigure
                });
            }else{
                console.log("empty dataset")
            }
        })
        await client.close();
        return result;

    } catch (e) {
        console.error(e);
        return false;
    }
}

async function debtRatio(request) {
    try {
        const uri = "mongodb://localhost:27017/";
        const client = new MongoClient(uri);
        await client.connect();

        const database = client.db("kennzahlen")
        const query = {"company_id": request.company_id};
        let data = database.collection('report').find(query)

        const result = {
            "company_id": request.company_id,
            "debt_ratios": []
        };

        await data.forEach((dataset) => {
            if (dataset){
                const stDebt = dataset.balance_sheet.passives.debt.short_term.liabilities;
                const ltDebt = dataset.balance_sheet.passives.debt.long_term.total;
                const fixedAssetsTotal = dataset.balance_sheet.actives.fixed_assets.total;
                const currentAssetsTotal = dataset.balance_sheet.actives.current_assets.total;

                const keyFigure = (stDebt + ltDebt) / (currentAssetsTotal + fixedAssetsTotal);

                result.debt_ratios.push({
                    "period": dataset.period,
                    "key_figure": keyFigure
                });
            }else{
                console.log("empty dataset")
            }
        })
        await client.close();
        return result;

    } catch (e) {
        console.error(e);
        return false;
    }
}

async function printResults(){
    console.log(await EquityRatio({company_id: "12345"}));
    console.log(await debtRatio({company_id: "12345"}));
}

printResults();
